### PR TITLE
Move pnp sample code into own folder

### PR DESCRIPTION
**Pull request by danewalton**
As per @ericwol-msft wishes. The main folder was getting pretty clogged.

### CODE REVIEW

File: sdk/samples/iot/pnp/pnp_protocol.c [hunk 1]
```diff
@@ -1,9 +1,9 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
-#include "iot_sample_common.h"
+#include "../iot_sample_common.h"
```
Comment (momuno) [inline sdk/samples/iot/pnp/pnp_protocol.c:4]: I'm curious if we can update cmake to recognize a path like "samples/iot/iot_sample_common.h".  While "../" works, I don't normally see that used in an include.

File: sdk/samples/iot/pnp/pnp_thermostat_component.c [hunk 2]
```diff
@@ -7,11 +7,11 @@
...
 
-#include "iot_sample_common.h"
-#include "sample_pnp.h"
-#include "sample_pnp_mqtt_component.h"
+#include "../iot_sample_common.h"
+#include "pnp_protocol.h"
+#include "pnp_mqtt_message.h"
 
-#include "sample_pnp_thermostat_component.h"
+#include "pnp_thermostat_component.h"
```
Comment (momuno) [inline sdk/samples/iot/pnp/pnp_thermostat_component.c:15]: Loving the directory and name updates.  Concise and organized!

File: sdk/samples/iot/pnp/pnp_protocol.c (hunk 1 above)
Comment (danewalton) [reply sdk/samples/iot/pnp/pnp_protocol.c]: good point. updated feel free to check

File: sdk/samples/iot/pnp/pnp_protocol.c [hunk 3]
```diff
@@ -1,9 +1,9 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
-#include "iot_sample_common.h"
+#include <iot_sample_common.h>
```
Comment (momuno) [inline sdk/samples/iot/pnp/pnp_protocol.c:4]: Good eye.
I was reading up on the difference, and my understanding is that "" means to check current directory (or sub directories), while <> means to check via a directory list determined by the compiler. Does that sound accurate?

Comment (danewalton) [reply sdk/samples/iot/pnp/pnp_protocol.c:4]: yea IIRC it has to do with precedence of lookup. `""` is local directory first while I believe `<>` doesn't look up locally.

**Pull request merged**